src/common/CommandLine.cs

```diff
@@ -590,11 +590,18 @@ public bool LastQualifierWins
 #endif
         public CommandLineParser(string commandLine)
         {
+            _args = new List<string>();
             ParseWords(commandLine);
         }
         public CommandLineParser(string[] args)
         {
-            _args = new List<string>(args);
+            _args = new List<string>();
+            _seenExeArg = true;
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                ParseWords(args[i]);
```

Comment:
Isn't this double-parsring the args? Suppose you had nested quotes with something like `"\"Hello\" \"World\""`.  Previously you'd get a single _arg entry as `"Hello" "World"`.  Now you'll get two entries: `Hello` and `World`.